dashboard/app/models/user.rb

```diff
@@ -2261,6 +2261,42 @@ def providers
     end
   end
 
+  # The number of times a user has attempted to join a section in the last 24 hours
+  # Returns nil if not section join attempts.
+  def num_failed_section_attempts
+    properties['section_attempts']
+  end
+
+  # There are two possible states in which we would want to reset the section attempt values
+  # 1) User has never joined a section and thus the values must be initialized
+  # 2) 24 hours have passed since the user last attempted to join a section
+  def reset_section_attempts?
+    !num_failed_section_attempts || (DateTime.now - DateTime.parse(properties['section_attempts_last_reset'])).to_i > 0
+  end
+
+  def reset_failed_section_attempts
+    properties['section_attempts'] = 0
+    properties['section_attempts_last_reset'] = DateTime.now.to_s
+    save(validate: false)
+  end
+
+  # Users must complete a captcha when joining a section if they have joined 3 or more in the past 24 hours
+  def display_captcha?
+    if reset_section_attempts?
+      reset_failed_section_attempts
+    end
+    num_failed_section_attempts >= 3
+  end
+
+  # Failed section attemps reset every day
```

Comment:
NIT: This header comment seems to describe an implementation detail. It may make more sense right before line 2292 to describe the if statement. 

Reasoning: header comments generally describe the overall purpose and/or functionality of a method. The way it's currently written makes me think the purpose of `increment_section_attempts` is to reset the failed section attempts count each day.